### PR TITLE
DOCSP-43063 Time Series Data

### PR DESCRIPTION
**Pull request by jordan-smith721**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-43063
Staging - https://deploy-preview-55--docs-c.netlify.app/databases-collections/time-series/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/databases-collections/time-series.txt [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+   To create and interact with time series collections, you must be
+   connected to a deployment running {+mdb-server+} 5.0 or later.
+
+You can create a time series collection to store time series data.  
+To create a time series collection, pass the following parameters to the 
+``mongoc_database_create_collection()`` method:
+
+- The database in which to create the collection
+- The name of the new collection to create
+- A ``timeseries`` object that specifies the ``timeField`` option
```
Comment (mcmorisi) [inline source/databases-collections/time-series.txt:72]: I'm not sure if "object" is the correct terminology here.

@kevinAlbs might be able to clarify.

File: source/databases-collections/time-series.txt [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+.. literalinclude:: /includes/databases-collections/timeseries.c
+   :language: c
+   :start-after: start-create-time-series
+   :end-before: end-create-time-series
+   :dedent:
+   :emphasize-lines: 13-19
+
+To verify that you successfully created the time series collection, run
+the ``mongoc_database_find_collections_with_opts()`` method on the
```
Comment (mcmorisi) [inline source/databases-collections/time-series.txt:87]:
```suggestion
the ``mongoc_database_find_collections_with_opts()`` function on the
```

File: source/databases-collections/time-series.txt [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      { "name" : "october2024", "type" : "timeseries", "options" : { "timeseries" : { "timeField" : "timestamp", "granularity" : "seconds", "bucketMaxSpanSeconds" : { "$numberInt" : "3600" } } }, "info" : { "readOnly" : false } }
+      ...
+
+.. _c-time-series-store:
+
+Store Time Series Data
+----------------------
+
+You can insert data into a time series collection by using the ``mongoc_collection_insert_one()``
+or ``mongoc_collection_insert_many()`` methods and specifying the measurement, timestamp, and metadata
```
Comment (mcmorisi) [inline source/databases-collections/time-series.txt:111]:
```suggestion
or ``mongoc_collection_insert_many()`` functions and specifying the measurement, timestamp, and metadata
```

File: source/databases-collections/time-series.txt [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+To learn more about performing read operations, see :ref:`c-read`.
+
+To learn more about performing aggregation operations, see the :ref:`c-aggregation`
+guide.
+
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+To learn more about the methods mentioned in this guide, see the following
```
Comment (mcmorisi) [inline source/databases-collections/time-series.txt:164]:
```suggestion
To learn more about the functions mentioned in this guide, see the following
```

File: source/databases-collections/time-series.txt [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+-------------------------------
+
+.. important:: Server Version for Time Series Collections
+
+   To create and interact with time series collections, you must be
+   connected to a deployment running {+mdb-server+} 5.0 or later.
+
+You can create a time series collection to store time series data.  
+To create a time series collection, pass the following parameters to the 
+``mongoc_database_create_collection()`` method:
```
Comment (mcmorisi) [inline source/databases-collections/time-series.txt:68]:
```suggestion
``mongoc_database_create_collection()`` function:
```

**Review by mcmorisi — CHANGES_REQUESTED** [review]
Some nits and one open question! Looks good otherwise, but would appreciate @kevinAlbs feedback on one point.

File: source/databases-collections/time-series.txt (hunk 2 above)
Comment (jordan-smith721) [reply source/databases-collections/time-series.txt]: Oops, good catch

**Review by mcmorisi — APPROVED** [review]
LGTM!

File: source/includes/databases-collections/timeseries.c [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    const bson_t *insert_doc;
+    insert_doc = BCON_NEW (
+        "temperature", BCON_DOUBLE (70.0),
+        "location", "{", "city", BCON_UTF8 ("New York"), "}",
+        "timestamp", BCON_DATE_TIME (1633046400000));
+
+    mongoc_collection_insert_one (collection, insert_doc, NULL, NULL, &error))
+    // end-insert-document
+
+    bson_destroy (opts);
```
Comment (joshbsiegel) [inline source/includes/databases-collections/timeseries.c:60]:
```suggestion
```

File: source/includes/databases-collections/timeseries.c [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+    // Create options for the time series collection
+    bson_t *opts = BCON_NEW (
+        "timeseries", "{",
+        "timeField", BCON_UTF8 ("timestamp"),
+        "}");
+
+    // Create the time series collection
+    mongoc_database_create_collection (database, "october2024", opts, &error);
+    // end-create-time-series
```
Comment (joshbsiegel) [inline source/includes/databases-collections/timeseries.c:28]:
```suggestion
    bson_destroy (opts);
    // end-create-time-series
```

File: source/includes/databases-collections/timeseries.c [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        "location", "{", "city", BCON_UTF8 ("New York"), "}",
+        "timestamp", BCON_DATE_TIME (1633046400000));
+
+    mongoc_collection_insert_one (collection, insert_doc, NULL, NULL, &error))
+    // end-insert-document
+
+    bson_destroy (opts);
+    mongoc_cursor_destroy (cursor);
+    mongoc_database_destroy (database);
+    mongoc_client_destroy (client);
```
Comment (joshbsiegel) [inline source/includes/databases-collections/timeseries.c:63]:
```suggestion
    mongoc_client_destroy (client);
    mongoc_collection_destroy (collection);
```

File: source/includes/databases-collections/timeseries.c [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    insert_doc = BCON_NEW (
+        "temperature", BCON_DOUBLE (70.0),
+        "location", "{", "city", BCON_UTF8 ("New York"), "}",
+        "timestamp", BCON_DATE_TIME (1633046400000));
+
+    mongoc_collection_insert_one (collection, insert_doc, NULL, NULL, &error))
+    // end-insert-document
+
+    bson_destroy (opts);
+    mongoc_cursor_destroy (cursor);
```
Comment (joshbsiegel) [inline source/includes/databases-collections/timeseries.c:61]:
```suggestion
```

File: source/includes/databases-collections/timeseries.c [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        char *str = bson_as_canonical_extended_json (doc, NULL);
+        printf ("%s\n", str);
+        bson_free (str);
+    }
+
+    // Check for cursor errors
+    if (mongoc_cursor_error (cursor, &error))
+    {
+        fprintf (stderr, "Cursor error: %s\n", error.message);
+    }
```
Comment (joshbsiegel) [inline source/includes/databases-collections/timeseries.c:47]:
```suggestion
    }
    mongoc_cursor_destroy (cursor);
```

**Review by joshbsiegel — CHANGES_REQUESTED** [review]

File: source/databases-collections/time-series.txt (hunk 1 above)
Comment (joshbsiegel) [reply source/databases-collections/time-series.txt:72]: That particular parameter is a `bson_t`, so I would suggest altering as such:
```suggestion
- A BSON document that specifies the ``timeField`` option
```

**Review by joshbsiegel — CHANGES_REQUESTED** [review]

**Review by joshbsiegel — APPROVED** [review]
LGTM

**Pull request merged**